hybridse/src/node/node_manager.cc

```diff
@@ -753,15 +753,16 @@ SqlNode *NodeManager::MakeCreateIndexNode(const std::string &index_name,
     return RegisterNode(node_ptr);
 }
 
-DeployNode *NodeManager::MakeDeployStmt(const std::string &name, const SqlNode *stmt,
-                                     const std::string& stmt_str, bool if_not_exist) {
-    DeployNode *node = new DeployNode(name, stmt, stmt_str, if_not_exist);
+DeployNode *NodeManager::MakeDeployStmt(const std::string &name, const SqlNode *stmt, const std::string &stmt_str,
+                                        const std::shared_ptr<OptionsMap> options, bool if_not_exist) {
+    DeployNode *node = new DeployNode(name, stmt, stmt_str, std::move(options), if_not_exist);
     return RegisterNode(node);
 }
 
 DeployPlanNode *NodeManager::MakeDeployPlanNode(const std::string &name, const SqlNode *stmt,
-                                                const std::string& stmt_str, bool if_not_exist) {
-    DeployPlanNode *node = new DeployPlanNode(name, stmt, stmt_str, if_not_exist);
+                                                const std::string &stmt_str, const std::shared_ptr<OptionsMap> options,
+                                                bool if_not_exist) {
+    DeployPlanNode *node = new DeployPlanNode(name, stmt, stmt_str, std::move(options), if_not_exist);
```

Comment:
from `DeployNode` -> `DeployPlanNode` should not `std::move`, or the `options_` in `DeployNode` will become null

Reply:
the `options` here is value-copied. should not affect the `options_` in `DeployNode`?

Reply:
Oh yeah, forget it is returned by `Options`, but since it is a temp value, so copy elision happens without `std::move` ?

Reply:
I think since `options` is named, copy elision may not apply here.

Reply:
understand